### ISSUE TITLE
createGlobalStyle not being auto formatted

### ISSUE THREAD

**Issue by mkvlrn** [styled-components/vscode-styled-components#175]
Anything inside another styled tag will be properly formatted, for example a `styled.div`. But a `createGlobaStyle` function will not have it's contents formatted. Color is still being applied, but formatting won't happen.

**Comment by danielmahon** [comment]
Same issue, not sure of any side-effects yet but as a workaround, I'm just wrapping the whole thing in the `css` function, which seems to re-enable the auto-formatting.

```ts
import { createGlobalStyle, css } from 'styled-components';

export default createGlobalStyle`${css`
  html,
  body,
  #root {
    padding: 0px;
    margin: 0px;
    width: 100vw;
    height: 100vh;
  }
`}`;
```

**Comment by shmilyoo** [comment]
> Same issue, not sure of any side-effects yet but as a workaround, I'm just wrapping the whole thing in the `css` function, which seems to re-enable the auto-formatting.
> 
> ```ts
> import { createGlobalStyle, css } from 'styled-components';
> 
> export default createGlobalStyle`${css`
>   html,
>   body,
>   #root {
>     padding: 0px;
>     margin: 0px;
>     width: 100vw;
>     height: 100vh;
>   }
> `}`;
> ```

it works

**Comment by alikadir** [comment]
> > Same issue, not sure of any side-effects yet but as a workaround, I'm just wrapping the whole thing in the `css` function, which seems to re-enable the auto-formatting.
> > ```ts
> > import { createGlobalStyle, css } from 'styled-components';
> > 
> > export default createGlobalStyle`${css`
> >   html,
> >   body,
> >   #root {
> >     padding: 0px;
> >     margin: 0px;
> >     width: 100vw;
> >     height: 100vh;
> >   }
> > `}`;
> > ```
> 
> it works

it not works currently 🙁

**Comment by alikadir** [comment]
flowing code works 💪

```js
import * as styled from 'styled-components';

export default styled.createGlobalStyle`
  body {
    margin: 0;
  }
  code {
    font-family: source-code-pro;
  }
`;

```

**Comment by adi518** [comment]
@alikadir That's the best solution thus far.

**Comment by gurmiguel** [comment]
With typescript, if I try to pass the props type definition, createGlobalStyle loses the formatting

```ts
const GlobalStyles = createGlobalStyles<{prop: any}>`
  body {
    ...
  }
`
```

without the props type the formatting works normally

**Comment by jasonwilliams** [comment]
fixed in 1.0.0

**Comment by webdevelukas** [comment]
Doesn't work for me in VS-Code (1.50.1) on macOS (10.15.7) and with vscode-styled-components (1.2). Actually I'm using the `${css``}` workaround.

**Comment by jasonwilliams** [comment]
@webdevelukas could you paste a snippet of code you would expect to work? (Without the workaround)

**Comment by lucasengel** [comment]
> @webdevelukas could you paste a snippet of code you would expect to work? (Without the workaround)

This won't get formatted:

```js
import { createGlobalStyle } from "styled-components"

const GlobalStyles = createGlobalStyle`
  * {box-sizing: border-box;
    margin: 0;    padding: 0;
  }
`

export default GlobalStyles

**Comment by Nytheneals** [comment]
I'm late to the party but this will also work fine

```js
import * as styled from 'styled-components';

const GlobalStyles = styled.createGlobalStyle`
  * {
    box-sizing: border-box;
    margin: 0;
    padding: 0;
  }
`;

export default GlobalStyles;
```

**Comment by ibqn** [comment]
This bug ist still not addressed, why is issue marked as resolved?

The following block ist not formatted in vscode:

```jsx
import { createGlobalStyle } from 'styled-components'

const GlobalStyle =  createGlobalStyle`
  html,
  body {
    padding: 0px;
    margin: 0px;
    width: 100vw;
    height: 100vh;
  }
`
export default GlobalStyle
```

**Comment by Alissonsleal** [comment]
> This bug ist still not addressed, why is issue marked as resolved?
> 
> The following block ist not formatted in vscode:
> 
> ```js
> import { createGlobalStyle } from 'styled-components'
> 
> const GlobalStyle =  createGlobalStyle`
>   html,
>   body {
>     padding: 0px;
>     margin: 0px;
>     width: 100vw;
>     height: 100vh;
>   }
> `
> export default GlobalStyle
> ```

Try using this:
```js
import * as styled from 'styled-components';

const GlobalStyle =  styled.createGlobalStyle`
  html,
  body {
    padding: 0px;
    margin: 0px;
    width: 100vw;
    height: 100vh;
  }
`;

export default GlobalStyle;
```

**Comment by luizwhite** [comment]
> 
> 
> flowing code works 💪
> 
> ```js
> import * as styled from 'styled-components';
> 
> export default styled.createGlobalStyle`
>   body {
>     margin: 0;
>   }
>   code {
>     font-family: source-code-pro;
>   }
> `;
> ```

It bugs my Next.js when using styled-components, because **babel-plugin-styled-components** doesn't quite understand that.
So my workaworkaround is just type **_styled._** before **_createGlobalStyle_**, and my formatter already shows me the linting, and i can push my format/fix shortcut, then save the file after removing the typed **_styled._**

**Comment by ibqn** [comment]
@luizwhite in nextjs one can write something like this as a workaround 

```jsx
import { createGlobalStyle } from 'styled-components'

const styled = { createGlobalStyle }

const GlobalStyle = styled.createGlobalStyle`
  body {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
  }
  ...
`

export default GlobalStyle

```
but well I mean...

**Comment by jasonwilliams** [comment]
This is definitely working for me (as of v1.5.2), we even have it in our fixtures https://github.com/styled-components/vscode-styled-components/blob/27a0ab882628646d775989075cfb270085e011f2/src/tests/suite/colorize-fixtures/createGlobalStyle.js

Could someone confirm this is still broken and show their extension version?

**Comment by ibqn** [comment]
@jasonwilliams  no, it's still no working, i.e. code is colorized colorized, but styles are not formatted. 
I have even uninstalled and reinstalled this extension again, to double check.

```
Version: 1.55.2
Commit: 3c4e3df9e89829dce27b7b5c24508306b151f30d
Date: 2021-04-13T09:37:02.931Z
Electron: 11.3.0
Chrome: 87.0.4280.141
Node.js: 12.18.3
V8: 8.7.220.31-electron.0
OS: Linux x64 5.4.0-72-generic
```
![Screenshot from 2021-05-02 12-45-05](https://user-images.githubusercontent.com/234029/116810530-87d5e580-ab44-11eb-990d-28b0dc62d48f.png)
![Screenshot from 2021-05-02 12-44-43](https://user-images.githubusercontent.com/234029/116810534-91f7e400-ab44-11eb-8e39-cdad7e4ee8a0.png)
![Screenshot from 2021-05-02 12-48-47](https://user-images.githubusercontent.com/234029/116810611-fd41b600-ab44-11eb-9f0c-b34ce6742d78.png)

Feel free to let me know, if I somehow can assist here.

**Comment by jasonwilliams** [comment]
@ibqn can you show the list of extensions you have installed and what file type that file is?
Can you also paste the demo code from your screenshot here so i can try it aswell.

You may have something conflicting

**Comment by rensvis** [comment]
Issue for me was that multiple formatters were active and I just had to pick a default. You can do so by hitting Shift + Option + F and you'll get a window asking to select a default. Simply selecting Prettier here solved the non-formatting issue for me.

**Comment by dminkovsky** [comment]
> flowing code works 💪
> 
> ```js
> import * as styled from 'styled-components';
> 
> export default styled.createGlobalStyle`
>   body {
>     margin: 0;
>   }
>   code {
>     font-family: source-code-pro;
>   }
> `;
> ```

Why does this work?
